src/java/za/co/mip/ablduck/ABLDuck.java

```diff
@@ -0,0 +1,420 @@
+            try (FileWriter file = new FileWriter(outputFile.toString())) {
+                file.write("Ext.data.JsonP." + js.name.replace(".", "_") + "(" + gson.toJson(js) + ");");
+            } catch (IOException ex) {
+                throw new BuildException(ex);
+            }
+        }
+
+        File dataFile = new File(this.destDir, "data.js");
+        try (FileWriter file = new FileWriter(dataFile.toString())) {
+            file.write("Docs = {\"data\":" + gson.toJson(dataJSObject) + "}");
+        } catch (IOException ex) {
+            throw new BuildException(ex);
+        }
+        
+    }
+
+    private HierarchyResult determineClassHierarchy(SourceJSObject curClass, HierarchyResult result) {
+        
+        String inherits = curClass.ext;
+
+        if (!"".equals(inherits)) {
+            result.addHierarchy(inherits);
+
+            SourceJSObject nextClass = jsObjects.get(inherits);
+                        
+            if (nextClass != null) {
+                for (MemberObject member : nextClass.members) {
+                    if (member.owner.equals(inherits) && (member.meta.isPrivate == null || !member.meta.isPrivate)) 
+                        result.addInheritedmember(member);
+                }
+            
+                determineClassHierarchy(nextClass, result);
+            }
+        }
+        return result;
+    }
+
+    private void extractTemplateDirectory(File outputDir) throws IOException {
+ 
+        File template = exportResource("ablduck.zip", outputDir);
+
+        unzip(template, outputDir);
+
+        if (!template.delete()) {
+            log("Unable to delete " + template.getAbsolutePath() + ", please delete this manually.", Project.MSG_INFO);
+        }
+        
+ 
+    }
+
+    public File exportResource(String resourceName, File outputDir) throws IOException {
+        
+        
+        File resource = new File(outputDir, resourceName);
+
+        try (
+            InputStream stream = ABLDuck.class.getResourceAsStream("resources/" + resourceName);
+            OutputStream resStreamOut = new FileOutputStream(resource.getAbsolutePath());
+        ){
+            
+            if(stream == null) {
+                throw new IOException("Cannot get resource \"" + resourceName + "\" from Jar file.");
+            }
+ 
+            int readBytes;
+            byte[] buffer = new byte[BUFFER_SIZE];
+            
+            while ((readBytes = stream.read(buffer)) > 0) {
+                resStreamOut.write(buffer, 0, readBytes);
+            }
+        } catch (IOException ex) {
+            throw ex;
+        }
+    
+        return resource;
+    }
+    
+    public void unzip(File zipFile, File unzipTo) throws IOException {
+
+        if (!unzipTo.exists()) {
+            unzipTo.mkdir();
+        }
+        
+        
+        
+        try (
+            ZipInputStream zipIn = new ZipInputStream(new FileInputStream(zipFile.getAbsolutePath()));

```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this extraneous semicolon. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AS2959)